### ISSUE TITLE
How about `[]struct { Key string; Value *jsonschema.Schema }` instead of orderedmap?

### ISSUE THREAD

**Issue by chanced** [invopop/jsonschema#35]
I'm not knocking orderedmap but it doesn't enable inline definitions, which can be tedious if you want/need to create custom `Schema`s as it breaks the flow of the definition. 

Using 
```go
type Obj []struct {
	Field string
	Value *jsonschema.Schema
}
```
enables this sort of API:

```go
func (CreateTenantRequest) JSONSchema() *jsonschema.Schema {
	return &jsonschema.Schema{
		Type: "object",
		Properties: jsonschema.Obj{
			{"name", &jsonschema.Schema{}},
		},
	}
}
```

instead of:
```go
func (CreateTenantRequest) JSONSchema() *jsonschema.Schema {
	p := orderedmap.New()
	p.Set("name", &jsonschema.Schema{
		Type: "string",
	})
	return &jsonschema.Schema{
		Type: "object",
		Properties: p,
	}
}
```

Marshaling/Unmarshaling the JSON is incredibly straightforward with [tidwall/sjson](https://github.com/tidwall/sjson) & [tidwall/gjson](https://github.com/tidwall/gjson): 

https://go.dev/play/p/OxCa0Vyv6l4 (I used strings as the value to keep things simple / concise for demo purposes). Error checking and such was omitted as well.

Also, thanks for the package.

**Comment by samlown** [comment]
I'm inclined to agree that OrderedMap doesn't have a great interface, but in reality it is doing what you're suggesting with the `Obj` struct.

I don't have time to implement this now, but I'd suggest just creating a wrapper around the OrderedMap implementation, i.e. inside the `jsonschema` package:

```go
// Properties helps define an ordered map of property names to schemas
func Properties(in map[string]*Schema) *orderedmap.OrderedMap {
  m := orderedmap.New()
  for k, v := range in {
    m.Set(k, v)
  }
  return m
}
```

Then your schema could look like this, which isn't _quite_ as bad:

```go
func (CreateTenantRequest) JSONSchema() *jsonschema.Schema {
	return &jsonschema.Schema{
		Type: "object",
		Properties: jsonschema.Properties(map[string]*jsonschema.Schema{
			"name": &jsonschema.Schema{},
		}),
	}
}
```

**Comment by chanced** [comment]
> but in reality it is doing what you're suggesting with the Obj struct.

It is, but being a `struct`, it doesn't allow for inlining.

It's not that big of a deal. I created the issue (sorry, it should have been a thread in discussions) to see if you would be interested in it.

**Comment by chanced** [comment]
This is completely unrelated but the `Schema` should probably use the `https` scheme (`"http://json-schema.org/draft/2020-12/schema"`) as some implementations key off of it (because https is used in the spec / docs).

Fortunately `Schema` is configurable so it's no where near any sort of priority. I intended to create PR for that but my IDE butchered your formatting. Then I realized it was configurable and forgot to `sed` it.

I can create a new issue if you're interested or I can just open a PR.
